File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        vector<int>nums = nums1;
        int i = 0;
        int j = 0;
        int k = 0;
        while(i<m && j<n){
            if(nums[i]<=nums2[j]){

                    nums1[k++] = nums[i++];
                    

            }else{

                    nums1[k++] = nums2[j++];

            }
        }

            while(j<n){
                nums1[k++] = nums2[j++];
            }

            while(i<m){
                nums1[k++] = nums[i++];
            }

    }
        
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void swapgreater(int[] nums1, int[] nums2, int left, int right){
        if(nums1[left] > nums2[right]){
            int temp = nums1[left];
            nums1[left] = nums2[right];
            nums2[right] = temp;

        }
    }
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        if(n==0){
            return;
        }
        
        int N = m+n;
        int gap = (N / 2) + (N % 2);
                             
        while(gap > 0){
            int left = 0;
            int right = gap;
            
            while(right < N){
                if(left < m && right >= m){
                    swapgreater(nums1, nums2, left, right-m);
                }
                else if(left < m && right < m){
                    swapgreater(nums1, nums1, left, right);
                }
                else{
                    swapgreater(nums2, nums2, left - m, right - m);
                }
                left ++;
                right ++;
            }
            if(gap == 1) break;
            gap = (gap / 2) + (gap % 2);
        }
        for(int i = m; i < N; i++){
            nums1[i] = nums2[i - m];
        }
    }
}
